src/prototype_creep_routing.js

```diff
@@ -12,8 +12,16 @@ Creep.prototype.getRoute = function() {
   }
 
   let route = [];
+  let useHighWay = false;
+  switch (this.memory.role) {
+  case 'carry':
+  case 'claimer':
+  case 'nextroomer':
+    useHighWay = true;
+  }
```

Comment:
Not sure about this.

Other creeps like `sourcer` or `reserver` might need to route through highway rooms as well. So which creeps shouldn't move throw highway rooms? `repairer` and `universal`, ... - creeps which shouldn't route at all.